### PR TITLE
Display cloning progress

### PR DESCRIPTION
**Pull request by cbart**
Part of #39327

Screenshot 2023-04-03 at 8 43 03 PM

Repositories under site admin display cloning progress in the bottom line while cloning. The data is only queried when `clone-progress-logging` feature flag is on. This is important, as when the feature flag is off, the request for cloning progress goes straight to gitserver. With the feature flag on the back-end goes to the database to retrieve the data.

## Test plan

UI: Visual inspection.

Backend: Unit testing.

Loom also examining the feature flag:

https://www.loom.com/share/440c0fd88f6646079762a8dcfb4bdebc

### CODE REVIEW

**Comment by cbart** [comment]
This is only a preview draft. I need to ask Cody to add some tests.

File: client/web/src/site-admin/components/RepoMirrorInfo.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ export const RepoMirrorInfo: React.FunctionComponent<
                                 </Tooltip>
                             </>
                         )}
+                        {mirrorInfo.cloneInProgress && <> {mirrorInfo.cloneProgress || ''}</>}
```
Comment (mrnugget) [inline client/web/src/site-admin/components/RepoMirrorInfo.tsx:38]: I strongly feel that this should go into the next line and be wrapped in `<Code />` 😄 On the same line it looks noisy.

**Review by mrnugget — APPROVED** [review]
Code looks good to me, but I think it should go on another line :)

File: cmd/frontend/graphqlbackend/repository_mirror_test.go [hunk 2]
```diff
@@ -223,3 +226,118 @@ func TestCheckMirrorRepositoryRemoteURL(t *testing.T) {
...
+	db.UsersFunc.SetDefaultReturn(users)
+
+	backend.Mocks.Repos.GetByName = func(ctx context.Context, name api.RepoName) (*types.Repo, error) {
+		return &types.Repo{
+			Name:      "repo-name",
+			CreatedAt: time.Now(),
+			Sources:   map[string]*types.SourceInfo{"1": {}},
+		}, nil
+	}
+	defer func() {
```
[the remark this answers is not in the document]
Comment (cbart) [reply cmd/frontend/graphqlbackend/repository_mirror_test.go:262]: good catch

File: client/web/src/site-admin/components/RepoMirrorInfo.tsx (hunk 1 above)
Comment (cbart) [reply client/web/src/site-admin/components/RepoMirrorInfo.tsx]: Good point!

Junior designed Cezary is grateful for your keen eye and excellent taste gentlemen.

Check out the screenshot in the top of the PR description.

**Review by sashaostrikov — APPROVED** [review]
LGTM after your updates 👍🏻

**Review by mrnugget — APPROVED** [review]

**Pull request merged**